packages/plugin-ext-vscode/src/common/plugin-vscode-environment.ts

```diff
@@ -24,13 +24,32 @@ export class PluginVSCodeEnvironment {
     @inject(EnvVariablesServer)
     protected readonly environments: EnvVariablesServer;
 
-    protected _extensionsDirUri: URI | undefined;
-    async getExtensionsDirUri(): Promise<URI> {
-        if (!this._extensionsDirUri) {
+    protected _userExtensionsDirUri: URI | undefined;
+    protected _deployedPluginsUri: URI | undefined;
+    protected _tmpDirUri: URI | undefined;
+
+    async getUserExtensionsDirUri(): Promise<URI> {
+        if (!this._userExtensionsDirUri) {
             const configDir = new URI(await this.environments.getConfigDirUri());
-            this._extensionsDirUri = configDir.resolve('extensions');
+            this._userExtensionsDirUri = configDir.resolve('extensions');
         }
-        return this._extensionsDirUri;
+        return this._userExtensionsDirUri;
     }
 
+    async getDeploymentDirUri(): Promise<URI> {
+        if (!this._deployedPluginsUri) {
+            const configDir = new URI(await this.environments.getConfigDirUri());
+            this._deployedPluginsUri = configDir.resolve('deployedPlugins');
+        }
+        return this._deployedPluginsUri;
+    }
+
+    async getTempDirUri(prefix?: string): Promise<URI> {
+        if (!this._tmpDirUri) {
+            const configDir: URI = new URI(await this.environments.getConfigDirUri());
+            const processId: string = process.pid.toString();
+            this._tmpDirUri = configDir.resolve('tmp').resolve(`${prefix ?? 'tmp-'}-${processId}`);
```

Comment:
If I understood this correctly, this will use the operating system's temp dir mechanism, that we considered unsafer than a location within the user's conf dir.

Reply:
I'll try.